onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc

```diff
@@ -196,10 +196,19 @@ NnapiExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_view
     result.push_back(onnxruntime::make_unique<ComputeCapability>(std::move(sub_graph)));
   }
 
-  LOGS_DEFAULT(INFO) << "NnapiExecutionProvider::GetCapability,"
-                     << " number of partitions supported by NNAPI: " << result.size()
-                     << " number of nodes in the graph: " << graph_view.NumberOfNodes()
-                     << " number of nodes supported by NNAPI: " << num_of_supported_nodes;
+  auto num_of_partitions = result.size();
+  const auto summary_msg = MakeString(
+      "NnapiExecutionProvider::GetCapability,",
+      " number of partitions supported by NNAPI: ", num_of_partitions,
+      " number of nodes in the graph: ", graph_view.NumberOfNodes(),
+      " number of nodes supported by NNAPI: ", num_of_supported_nodes);
+
+  // If the graph is partitioned in multiple subgraphs, and this may impact performance,
+  // we want to give users a summary message at warning level.
+  if (num_of_partitions > 1)
```

Comment:
We can log warning message when num_of_partitions > certain number, 1 might not be the best choice here